android/src/main/kotlin/io/flutter/plugins/geofencing/GeofencingPlugin.kt

```diff
@@ -178,17 +178,33 @@ class GeofencingPlugin(context: Context, activity: Activity?) : MethodCallHandle
             val ids = listOf(args!![0] as String)
             geofencingClient.removeGeofences(ids).run {
                 addOnSuccessListener {
-                    result.success(true)
                     for (id in ids) {
                         removeGeofenceFromCache(context, id)
                     }
+                    result.success(true)
                 }
                 addOnFailureListener {
                     result.error(it.toString(), null, null)
                 }
             }
         }
 
+        @JvmStatic
+        private fun getRegisteredGeofenceIds(context: Context, result: Result) {
+            synchronized(sGeofenceCacheLock) {
+                var p = context.getSharedPreferences(SHARED_PREFERENCES_KEY, Context.MODE_PRIVATE)
+                var persistentGeofences = p.getStringSet(PERSISTENT_GEOFENCES_IDS, null)
+                if (persistentGeofences == null || persistentGeofences.size == 0) {
+                    return
```

Comment:
This should return an empty list via `result.success`, otherwise the Dart `Future` will never complete and the application will hang when there's no registered geofences and `getRegisteredGeofenceIds` is called.

Reply:
Done. I added changes as per your comments.